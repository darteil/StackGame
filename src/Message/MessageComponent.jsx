import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './styles.css';

export default class MessageComponent extends Component {
  constructor(props) {
    super(props);

    this.timeoutVariable = setTimeout(() => {
      this.props.hideMessage();
    }, 10000);
  }

  componentWillUnmount() {
    clearTimeout(this.timeoutVariable);
  }

  render() {
    return (
      <div className={styles.message}>
        <em onClick={this.props.hideMessage} />
        <p>{this.props.text}</p>
      </div>
    );
  }
}

MessageComponent.propTypes = {
  text: PropTypes.string.isRequired,
  hideMessage: PropTypes.func.isRequired
};
