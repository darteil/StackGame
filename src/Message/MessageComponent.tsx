import React, { useEffect } from 'react';
import styles from './styles.css';

interface IProps {
  text: string;
  hideMessage: () => void;
}

const MessageComponent = (props: IProps) => {
  let timeoutVariable: NodeJS.Timeout;

  useEffect(() => {
    timeoutVariable = setTimeout(() => {
      props.hideMessage();
    }, 10000);

    return () => clearTimeout(timeoutVariable);
  }, []);

  return (
    <div className={styles.message}>
      <em onClick={props.hideMessage} />
      <p>{props.text}</p>
    </div>
  );
};

export default MessageComponent;
