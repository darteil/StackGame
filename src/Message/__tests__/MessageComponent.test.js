import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import MessageComponent from 'App/Message/MessageComponent';


describe('Message component', () => {
  it('should render correctly', () => {
    const text = 'text example';
    const handleHide = jest.fn();

    const output = shallow(
      <MessageComponent hideMessage={handleHide} text={text} />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
