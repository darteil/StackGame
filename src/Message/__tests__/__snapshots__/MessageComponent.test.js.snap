// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component should render correctly 1`] = `
<div
  className="message"
>
  <em
    onClick={[MockFunction]}
  />
  <p>
    text example
  </p>
</div>
`;
